packages/opencensus-instrumentation-http/src/http.ts

```diff
@@ -17,88 +17,121 @@
 import {types} from '@opencensus/opencensus-core';
 import {classes} from '@opencensus/opencensus-core';
 import {logger} from '@opencensus/opencensus-core';
+import * as http from 'http';
 import * as semver from 'semver';
 import * as shimmer from 'shimmer';
 import * as url from 'url';
 import * as uuid from 'uuid';
 
-// import * as Debug from 'debug';
-// const debug = Debug('opencensus');
 
+// TODO: maybe we should have a setup as a Client or as Server.
+
+type HttpModuleType = typeof http;
+type RequestFunction = typeof http.request;
+
+/** Http instrumentation plugin for Opencensus */
 export class HttpPlugin extends classes.BasePlugin {
+  /**
+   * Atributes Names according to Opencensus HTTP Specs
+   * https://github.com/census-instrumentation/opencensus-specs/blob/master/trace/HTTP.md
+   */
+  readonly ATRIBUTE_HTTP_HOST = 'http.host';
+  readonly ATRIBUTE_HTTP_METHOD = 'http.method';
+  readonly ATRIBUTE_HTTP_PATH = 'http.path';
+  readonly ATRIBUTE_HTTP_ROUTE = 'http.route';
+  readonly ATRIBUTE_HTTP_USER_AGENT = 'http.user_agent';
+  readonly ATRIBUTE_HTTP_STATUS_CODE = 'http.status_code';
+  // NOT ON OFFICIAL SPEC
+  readonly ATRIBUTE_HTTP_ERROR_NAME = 'http.error_name';
+  readonly ATRIBUTE_HTTP_ERROR_MESSAGE = 'http.error_mensage';
+
   logger: types.Logger;
 
   /** Constructs a new HttpPlugin instance. */
   constructor(moduleName: string) {
     super(moduleName);
   }
 
+
   /**
    * Patches HTTP incoming and outcoming request functions.
-   * @param moduleExporters The HTTP package.
+   * @param moduleExports The http module exports
    * @param tracer A tracer instance to create spans on.
    * @param version The package version.
    */
-  // tslint:disable:no-any
-  applyPatch(moduleExporters: any, tracer: types.Tracer, version: string) {
-    this.setPluginContext(moduleExporters, tracer, version);
+  // disable-next-line to disable no-any check for moduleExporters parameter
+  // tslint:disable-next-line
+  applyPatch(moduleExports: any, tracer: types.Tracer, version: string) {
```

Comment:
`moduleExports: HttpModule`

Then you can also remove the disable tslint lines.

Reply:
Some side effects on the compilation. A TODO was added.